README.rst

```diff
@@ -1,66 +1,202 @@
-Welcome to drf_polymorphic's documentation!
-=================================================
+drf_polymorphic
+===============
 
 :Version: 0.1.0
 :Source: https://github.com/maykinmedia/drf_polymorphic
 :Keywords: django, rest, polymorphic
-:PythonVersion: 3.7
+:PythonVersion: 3.7, 3.8
 
-|build-status| |coverage| |black|
+|build-status| |coverage| |linting|
 
-|python-versions| |django-versions| |pypi-version|
+Polymorphic support for DRF without Django models
 
-Polymorphic serializers without models
+Overview
+========
 
-.. contents::
+Polymorphism happens when a resource takes a certain shape depending on the type
+of the resource. Usually they have a common base type. The exact type/shape is not
+statically known, but depends on the run-time values.
 
-.. section-numbering::
+Unlike `django-rest-polymorphic <https://github.com/apirobot/django-rest-polymorphic>`_
+this implementation doesn't require the usage of ``django-polymorphic`` Model
+and can be used even without django models.
 
-Features
-========
+The implementations also includes the extension for ``drf_spectacular`` schema generation.
 
-* ...
-* ...
+``drf_polymorphic`` is inspired on the vng-api-common `implementation
+<https://github.com/VNG-Realisatie/vng-api-common/blob/master/vng_api_common/polymorphism.py>`_
 
 Installation
 ============
+Install using pip:
 
-Requirements
-------------
-
-* Python 3.7 or above
-* setuptools 30.3.0 or above
-* Django 2.2 or newer
+.. code-block:: bash
 
+    pip install drf_polymorphic
 
-Install
--------
+Then add ``drf_polymorphic`` to installed apps in ``settings.py``:
 
-.. code-block:: bash
+.. code-block:: python
 
-    pip install drf_polymorphic
+    INSTALLED_APPS = [
+        # ALL YOUR APPS
+        'drf_polymorphic',
+    ]
 
 
 Usage
 =====
 
-<document or refer to docs>
+For example, you have data for pets with structure dependent on the type of the pet:
 
+.. code-block:: python
 
+    from dataclasses import dataclass
 
-.. |build-status| image:: https://travis-ci.org/maykinmedia/drf_polymorphic.svg?branch=master
-    :target: https://travis-ci.org/maykinmedia/drf_polymorphic
 
-.. |coverage| image:: https://codecov.io/gh/maykinmedia/drf_polymorphic/branch/master/graph/badge.svg
-    :target: https://codecov.io/gh/maykinmedia/drf_polymorphic
-    :alt: Coverage status
+    @dataclass
+    class Pet:
+        name: str
+
+
+    @dataclass
+    class Cat(Pet):
+        hunting_skill: str
+        pet_type: str = 'cat'
```

Comment:
```suggestion
        pet_type = 'cat'
```
leaving off the type hint makes it a class attribute that cannot be specified in the `__init__`. Applies for all subclasses.